gremlin-dotnet/src/Gremlin.Net/Driver/ConnectionPool.cs

```diff
@@ -131,7 +131,15 @@ private Connection TryGetAvailableConnection(Connection[] connections)
                 }
                 return connection;
             }
-            throw new ConnectionPoolBusyException(_poolSize, _maxInProcessPerConnection);
+
+            if (connections.Length > 0) 
```

Comment:
This improvement definitely also makes sense, but it is not really related to TINKERPOP-2192. So, could you please create a different PR and ideally an issue in JIRA for this?